### PR TITLE
feat(button): add XL size variant

### PR DESCRIPTION
**Pull request by kennylam**
### Related Ticket(s)

#526 

### Description

Adds extra large size variant

### Changelog

**New**

- `xl` size variant

### CODE REVIEW

File: src/components/button/button.ts [hunk 1]
```diff
@@ -70,6 +70,11 @@ export enum BUTTON_SIZE {
    */
   SMALL = 'sm',
 
+  /**
+   * X-Large size.
+   */
+  XLARGE = 'xl',
```
Comment (asudoh) [inline src/components/button/button.ts:76]: To align to: https://github.com/carbon-design-system/carbon-web-components/blob/v1.8.0/src/components/dropdown/dropdown.ts#L82and https://github.com/carbon-design-system/carbon-web-components/blob/v1.8.0/src/components/input/input.ts#L46

```suggestion
  EXTRA_LARGE = 'xl',
```

File: src/components/button/button-story.ts [hunk 2]
```diff
@@ -33,6 +33,7 @@ const kinds = {
 const sizes = {
   'Regular size': null,
   [`Small size (${BUTTON_SIZE.SMALL})`]: BUTTON_SIZE.SMALL,
+  [`XL size (${BUTTON_SIZE.XLARGE})`]: BUTTON_SIZE.XLARGE,
```
Comment (asudoh) [inline src/components/button/button-story.ts:36]:
```suggestion
  [`XL size (${BUTTON_SIZE.XLARGE})`]: BUTTON_SIZE.EXTRA_LARGE,
```

**Review by asudoh — COMMENTED** [review]
Thank you for jumping in @kennylam!

File: src/components/button/button.ts (hunk 1 above)
Comment (kennylam) [reply src/components/button/button.ts]: Thanks @asudoh!

File: src/components/button/button-story.ts (hunk 2 above)
Comment (kennylam) [reply src/components/button/button-story.ts]: Thanks @asudoh!

**Review by asudoh — APPROVED** [review]
Thanks @kennylam for your quick update!

**Pull request merged**